pkg/ingester/transfer.go

```diff
@@ -284,6 +284,30 @@ func (i *Ingester) TransferTSDB(stream client.Ingester_TransferTSDBServer) error
 		receivedFiles.Add(float64(filesXfer))
 		level.Error(util.Logger).Log("msg", "Total files xfer", "from_ingester", fromIngesterID, "num", filesXfer)
 
+		// At this point all TSDBs have been received, so we can proceed loading TSDBs in memory.
+		// This is required because of two reasons:
+		// 1. No WAL replay performance penalty once the ingester switches to ACTIVE state
+		// 2. If a query is received on user X, for which the TSDB has been transferred, before
+		//    the first series is ingested, if we don't open the TSDB the query will return an
+		//    empty result (because the TSDB is opened only on first push or transfer)
+		userIDs, err := ioutil.ReadDir(i.cfg.TSDBConfig.Dir)
+		if err != nil {
+			return errors.Wrap(err, fmt.Sprintf("unable to list TSDB users in %s", i.cfg.TSDBConfig.Dir))
+		}
+
+		for _, user := range userIDs {
+			userID := user.Name()
+
+			level.Info(util.Logger).Log("msg", fmt.Sprintf("Loading TSDB for user %s", userID))
+			_, err = i.getOrCreateTSDB(userID, true)
```

Comment:
Do we maybe want to do this concurrently?

Reply:
I agree, but I would suggest to keep it for a dedicated PR (it's in my backlog) in order to keep this PR a bit easier (running it in parallel means we need a queue to limit concurrent jobs, ideally let configure the queue size, etc... so some further code will be required).

Reply:
Sounds good